src/veritysetup/veritysetup.c

```diff
@@ -109,11 +121,63 @@ static int parse_options(const char *options) {
                 else if (streq(word, "panic-on-corruption"))
                         arg_activate_flags |= CRYPT_ACTIVATE_PANIC_ON_CORRUPTION;
 #endif
-                else if ((val = startswith(word, "hash-offset="))) {
+                else if ((val = startswith(word, "no-superblock="))) {
+
+                        r = parse_boolean(val);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to parse %s, ignoring: %m", word);
+
+                        arg_no_superblock = r;
+                } else if ((val = startswith(word, "format="))) {
+
+                        r = safe_atoi(val, &arg_format);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to parse %s, ignoring: %m", word);
+                } else if ((val = startswith(word, "data-block-size="))) {
+
+                        r = safe_atou64(val, &arg_data_block_size);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to parse %s, ignoring: %m", word);
+                } else if ((val = startswith(word, "hash-block-size="))) {
+
+                        r = safe_atou64(val, &arg_hash_block_size);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to parse %s, ignoring: %m", word);
+                } else if ((val = startswith(word, "data-blocks="))) {
+
+                        r = safe_atou64(val, &arg_data_blocks);
+                        if (r < 0)
+                                log_error_errno(r, "Failed to parse %s, ignoring: %m", word);
+                } else if ((val = startswith(word, "hash-offset="))) {
 
                         r = safe_atou64(val, &arg_hash_offset);
                         if (r < 0)
                                 log_error_errno(r, "Failed to parse %s, ignoring: %m", word);
+                } else if ((val = startswith(word, "salt="))) {
+
+                        if (!*val) {
+                                free(arg_salt);
+                                arg_salt = NULL;
+                                arg_salt_size = 32;
+                        } else if (streq(val, "-")) {
+                                free(arg_salt);
+                                arg_salt = NULL;
+                                arg_salt_size = 0;
+                        } else {
+                                r = unhexmem(val, strlen(val), &arg_salt, &arg_salt_size);
```

Comment:
unhexmem() can fail for various reasons, not just OOM, for example because you pass invalid data such as "öäüß", please generate a proper log message